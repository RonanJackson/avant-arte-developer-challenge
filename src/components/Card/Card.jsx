import React, { useEffect, useState } from 'react';
import STYLES from './Card.scss';
import PropTypes from 'prop-types';

const getClassName = (className) => STYLES[className] || 'UNKNOWN';
const internationalNumberFormat = new Intl.NumberFormat('en-EU');

const Card = ({ title, tags, image, url, price }) => {
  const [tagVisibility, setTagVisibility] = useState({ visibility: 'visible' });
  const [tagText, setTagText] = useState('-');

  useEffect(() => {
    if (tags.includes('Sold Out')) setTagText('Sold Out');
    else if (tags.includes('Coming soon')) setTagText('Coming Soon');
    else setTagVisibility({ visibility: 'hidden' });
  }, [tags]);

  return (
    <article
      className={getClassName('Card')}
      onClick={() => window.open(url, '_blank')}
    >
      <div className={getClassName('Card__status-header')}>
        <div className={getClassName('Card__status-box')} style={tagVisibility}>
          <span>{tagText}</span>
        </div>
      </div>
      <img src={image} alt={title} className={getClassName('Card__image')} />
      <h1>{title}</h1>
      <h2>â‚¬{internationalNumberFormat.format(price)}</h2>
    </article>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  tags: PropTypes.arrayOf(PropTypes.string),
  image: PropTypes.string,
  url: PropTypes.string,
  price: PropTypes.number,
};

export default Card;
