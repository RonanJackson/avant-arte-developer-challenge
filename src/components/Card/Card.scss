@import '../../styles/mixin.scss';
@import '../../styles/variables.scss';
.Card {
  border: 1px solid $light;
  box-sizing: border-box;
  flex-basis: 50%;
  height: 52vh;
  padding: 3vh;
  cursor: pointer;

  h1 {
    color: $storm;
  }

  h2 {
    color: $cloud;
    font-size: medium;
    font-weight: lighter;
    margin-top: -1vh;
  }

  &__image {
    height: 35vh;
    width: 25vh;
    object-fit: contain;
    padding: 1vh;
    margin-left: auto;
    margin-right: auto;
    display: block;
  }

  &__status-header {
    display: flex;
    justify-content: flex-end;
  }

  &__status-box {
    padding: 0.7vh;
    border: 1px solid $light;
    font-size: smaller;
  }
}
