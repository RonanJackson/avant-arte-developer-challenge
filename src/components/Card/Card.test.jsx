import React from 'react';
import TestRenderer from 'react-test-renderer';

import Card from './Card';

describe('Card', () => {
  it('should render correctly with no tag', () => {
    const tree = TestRenderer.create(
      <Card
        key="id123"
        title="title"
        tags={['Available']}
        image="//cdn.shopify.com/s/files/1/0260/5524/5923/products/SUSUMU_02copy_4488d8f1-6536-49ef-a310-adec11d2213d_500x.jpg?v=1613558884"
        url="https://shop.avantarte.com/products/the-sinner-at-dusk-1"
        price={4000.0}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with coming soon tag', () => {
    const tree = TestRenderer.create(
      <Card
        key="id123"
        title="title"
        tags={['Coming soon']}
        image="//cdn.shopify.com/s/files/1/0260/5524/5923/products/SUSUMU_02copy_4488d8f1-6536-49ef-a310-adec11d2213d_500x.jpg?v=1613558884"
        url="https://shop.avantarte.com/products/the-sinner-at-dusk-1"
        price={4000.0}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const tree = TestRenderer.create(
      <Card
        key="id123"
        title="title"
        tags={['Sold Out']}
        image="//cdn.shopify.com/s/files/1/0260/5524/5923/products/SUSUMU_02copy_4488d8f1-6536-49ef-a310-adec11d2213d_500x.jpg?v=1613558884"
        url="https://shop.avantarte.com/products/the-sinner-at-dusk-1"
        price={4000.0}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
