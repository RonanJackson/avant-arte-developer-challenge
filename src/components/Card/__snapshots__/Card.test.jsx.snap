// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<article
  className="Card"
  onClick={[Function]}
>
  <div
    className="Card__status-header"
  >
    <div
      className="Card__status-box"
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <span>
        -
      </span>
    </div>
  </div>
  <img
    alt="title"
    className="Card__image"
    src="//cdn.shopify.com/s/files/1/0260/5524/5923/products/SUSUMU_02copy_4488d8f1-6536-49ef-a310-adec11d2213d_500x.jpg?v=1613558884"
  />
  <h1>
    title
  </h1>
  <h2>
    €
    4,000
  </h2>
</article>
`;

exports[`Card should render correctly with coming soon tag 1`] = `
<article
  className="Card"
  onClick={[Function]}
>
  <div
    className="Card__status-header"
  >
    <div
      className="Card__status-box"
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <span>
        Coming Soon
      </span>
    </div>
  </div>
  <img
    alt="title"
    className="Card__image"
    src="//cdn.shopify.com/s/files/1/0260/5524/5923/products/SUSUMU_02copy_4488d8f1-6536-49ef-a310-adec11d2213d_500x.jpg?v=1613558884"
  />
  <h1>
    title
  </h1>
  <h2>
    €
    4,000
  </h2>
</article>
`;

exports[`Card should render correctly with no tag 1`] = `
<article
  className="Card"
  onClick={[Function]}
>
  <div
    className="Card__status-header"
  >
    <div
      className="Card__status-box"
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <span>
        -
      </span>
    </div>
  </div>
  <img
    alt="title"
    className="Card__image"
    src="//cdn.shopify.com/s/files/1/0260/5524/5923/products/SUSUMU_02copy_4488d8f1-6536-49ef-a310-adec11d2213d_500x.jpg?v=1613558884"
  />
  <h1>
    title
  </h1>
  <h2>
    €
    4,000
  </h2>
</article>
`;
